lib/core/src/Cardano/Wallet.hs

```diff
@@ -492,9 +497,33 @@ newWalletLayer tracer bp db nw tl = do
         fmap (\(PrimaryKey wid) -> wid) <$> DB.listWallets db
 
     _listTransactions
-        :: WalletId
-        -> ExceptT ErrNoSuchWallet IO [(Tx t, TxMeta)]
-    _listTransactions wid = liftIO $ map snd <$> DB.readTxHistory db (PrimaryKey wid)
+        :: (DefineTx t)
+        => WalletId
+        -> ExceptT ErrNoSuchWallet IO [TransactionInfo]
+    _listTransactions wid =
+        liftIO $ assemble <$> DB.readTxHistory db (PrimaryKey wid)
+      where
+        -- This relies on DB.readTxHistory returning all necessary transactions
+        -- to assemble coin selection information for outgoing payments.
+        -- To reliably provide this information, it should be looked up when
+        -- applying blocks, but that is future work.
```

Comment:
Just wondering, is there an issue that we can link to for this future work?

If not, then perhaps it would be worthwhile creating one, stating the problems with our current solution, and what we might do to improve?

Reply:
OK, created #573.

Reply:
Hmmm. We don't create issues outside of the iteration planning session actually :no_mouth: ! 
Future work goes on our roadmap, and is organized bi-monthly. This to avoid rotting issues and excessively big scope. 
Plus, there are still some details to figure out business-wise before working on that topic. I'll close / relocate #573